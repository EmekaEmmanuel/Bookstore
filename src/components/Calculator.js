import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Nav from './Nav';

function Calculator() {
  const initialState = {
    total: null,
    next: null,
    operation: null,
  };

  const [data, setData] = useState(initialState);

  const getButtonName = (e) => {
    const btnName = e.target.value;
    const result = calculate(data, btnName);
    setData(result);
  };

  return (
    <div className="app_container">
      <Nav />

      <div className="app_holder">
        <article>
          Let&apos;s do some Maths!.
        </article>

        <div className="app_box">
          <section className="result">
            <p>
              {data.total}
              {data.operation}
              {data.next}
            </p>
          </section>

          <section className="quad">
            <button value="A/C" type="button" onClick={getButtonName} className="normal_btn">A/C</button>
            <button value="%" type="button" onClick={getButtonName} className="normal_btn negative_margin">+/-</button>
            <button value="%" type="button" onClick={getButtonName} className="normal_btn negative_margin">%</button>
            <button value="รท" type="button" onClick={getButtonName} className="orange_btn negative_margin">รท</button>
          </section>

          <section className="quad">
            <button value="7" type="button" onClick={getButtonName} className="normal_btn">7</button>
            <button value="8" type="button" onClick={getButtonName} className="normal_btn negative_margin">8</button>
            <button value="9" type="button" onClick={getButtonName} className="normal_btn negative_margin">9</button>
            <button value="x" type="button" onClick={getButtonName} className="orange_btn negative_margin">x</button>
          </section>

          <section className="quad">
            <button value="4" type="button" onClick={getButtonName} className="normal_btn">4</button>
            <button value="5" type="button" onClick={getButtonName} className="normal_btn negative_margin">5</button>
            <button value="6" type="button" onClick={getButtonName} className="normal_btn negative_margin">6</button>
            <button value="-" type="button" onClick={getButtonName} className="orange_btn negative_margin">-</button>
          </section>

          <section className="quad">
            <button value="1" type="button" onClick={getButtonName} className="normal_btn">1</button>
            <button value="2" type="button" onClick={getButtonName} className="normal_btn negative_margin">2</button>
            <button value="3" type="button" onClick={getButtonName} className="normal_btn negative_margin">3</button>
            <button value="+" type="button" onClick={getButtonName} className="orange_btn negative_margin">+</button>
          </section>

          <section className="quad">
            <button value="0" type="button" onClick={getButtonName} className="normal_big_btn">0</button>
            <button value="." type="button" onClick={getButtonName} className="normal_btn negative_margin">.</button>
            <button value="=" type="button" onClick={getButtonName} className="orange_btn negative_margin">=</button>
          </section>
        </div>
      </div>

    </div>

  );
}

export default Calculator;
