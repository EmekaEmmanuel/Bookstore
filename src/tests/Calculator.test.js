import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Calculator />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
