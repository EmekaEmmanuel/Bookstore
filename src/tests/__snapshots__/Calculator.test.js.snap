// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app_container"
>
  <div
    className="nav_container"
  >
    <header
      className="nav_header"
    >
      <h1>
        Math Magicians
      </h1>
      <nav
        className="nav_list"
      >
        <li>
          <a
            className="nav_anchor td_none"
            href="/"
            onClick={[Function]}
          >
            Home 
          </a>
        </li>
        <li
          className="border_active"
        >
          <a
            className="nav_anchor  td_none"
            href="/Calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className="nav_anchor td_none"
            href="/Quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </nav>
    </header>
  </div>
  <div
    className="app_holder"
  >
    <article>
      Let's do some Maths!.
    </article>
    <div
      className="app_box"
    >
      <section
        className="result"
      >
        <p />
      </section>
      <section
        className="quad"
      >
        <button
          className="normal_btn"
          onClick={[Function]}
          type="button"
          value="A/C"
        >
          A/C
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="%"
        >
          +/-
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="orange_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
      </section>
      <section
        className="quad"
      >
        <button
          className="normal_btn"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="orange_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
      </section>
      <section
        className="quad"
      >
        <button
          className="normal_btn"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="orange_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
      </section>
      <section
        className="quad"
      >
        <button
          className="normal_btn"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="orange_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
      </section>
      <section
        className="quad"
      >
        <button
          className="normal_big_btn"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="normal_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="orange_btn negative_margin"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </section>
    </div>
  </div>
</div>
`;
